src/plugins/usage_collection/README.md

```diff
@@ -0,0 +1,139 @@
+# Kibana Usage Collection Service
+
+Usage Collection allows collecting usage data for other services to consume (telemetry and monitoring).
+To integrate with the telemetry services for usage collection of your feature, there are 2 steps:
+
+1. Create a usage collector.
+2. Register the usage collector.
+
+## Creating and Registering Usage Collector
+
+All you need to provide is a `type` for organizing your fields, and a `fetch` method for returning your usage data. Then you need to make the Telemetry service aware of the collector by registering it.
+
+### New Platform:
+
+1. Make sure `usageCollection` is in your optional Plugins:
+
+```json
+// plugin/kibana.json
+{
+  "id": "...",
+  "optionalPlugins": ["usageCollection"]
+}
+```
+
+2. Register Usage collector in the `setup` function:
+
+```ts
+// server/plugin.ts
+class Plugin {
+  setup(core, plugins) {
+    registerMyPluginUsageCollector(plugins.usageCollection);
+  }
+}
+```
+
+3. Creating and registering a Usage Collector. Ideally collectors would be defined in a separate director `server/collectors/register.ts`.
+
+```ts
+// server/collectors/register.ts
+import { PluginSetupContract as UsageCollection } from 'src/plugins/usage_collection/server';
+import { CallCluster } from 'src/legacy/core_plugins/elasticsearch';
+
+export function registerMyPluginUsageCollector(usageCollection?: UsageCollection): void {
+  // usageCollection is an optional dependency, so make sure to return if it is not registered.
+  if (!usageCollection) {
+    return;
+  }
+
+  // create usage collector
+  const myCollector = usageCollection.makeUsageCollector({
+    type: MY_USAGE_TYPE,
+    fetch: async (callCluster: CallCluster) => {
+
+    // query ES and get some data
+    // summarize the data into a model
+    // return the modeled object that includes whatever you want to track
+
+      return {
+        my_objects: {
+          total: SOME_NUMBER
+        }
+      };
+    },
+  });
+
+  // register usage collector
+  usageCollection.registerCollector(myCollector);
```

Comment:
Is this step useful? Seems like it'd be nice if `usageCollection.makeUsageCollector` just automatically registered it as well.

Reply:
I wanted to keep the changes to the `collectorSet` api to a minimal.

Initially the idea behind was that we might want to do other stuff to the collector other than registering it with the usage collector.

This might still be the case we'd want to have with `statsCollector`. which use the same collector but register it under `statsCollection`: used for monitoring ui stats and does not report it under `usage`.